plugins/permCheck.ts

```diff
@@ -19,21 +27,247 @@
  * @end
  */
 
-import type { GuildMember, PermissionResolvable } from "discord.js";
-import { CommandControlPlugin, CommandType, controller } from "@sern/handler";
-export function permCheck(perm: PermissionResolvable, response: string) {
-	return CommandControlPlugin<CommandType.Both>(async (ctx, args) => {
-		if (ctx.guild === null) {
-			await ctx.reply("This command cannot be used here");
-			console.warn(
-				"PermCheck > A command stopped because we couldn't check a users permissions (was used in dms)",
-			); //delete this line if you dont want to be notified when a command is used outside of a guild/server
-			return controller.stop();
-		}
-		if (!(ctx.member! as GuildMember).permissions.has(perm)) {
-			await ctx.reply(response);
-			return controller.stop();
-		}
-		return controller.next();
-	});
+import {
+  Colors,
+  EmbedBuilder,
+  PermissionsBitField,
+  type GuildMember,
+  type PermissionResolvable,
+  type TextChannel
+} from "discord.js";
+import {
+  CommandControlPlugin,
+  CommandType,
+  controller,
+  Service
+} from "@sern/handler";
+
+function command(perm: PermissionResolvable, response?: string) {
+  return CommandControlPlugin<CommandType.Both>(async (ctx) => {
+    if (ctx.guild === null) {
+      await ctx.reply("This command cannot be used in DM's!");
```

Comment:
maybe make these ephemeral? need the rest of the team's opinion

Reply:
I had two reasons for not making this ephemeral.
- It's not in a guild
- It's possibly a text command

Reply:
oh i see